File: LeetCode75/2390_Removing_Stars_From_a_String/2390_Removing_Stars_From_a_String.ts
function removeStars(s: string): string {
	let result: string[] = [];
	
	for (let i = 0; i < s.length; i++) {
		if (s[i] === '*') {
			// *が見つかった場合、結果配列から最後の文字を削除
			if (result.length > 0) {
				result.pop();
			}
		} else {
			// *以外の文字は結果配列に追加
			result.push(s[i]);
            console.log(result);
		}
	}
	
	// 配列を文字列に変換して返す
	return result.join('');
}

/**
【アプローチ】
*を見つけ出して、それより1つ前の文字を削除する
・宣言をする
・ループ処理をする
・if文で、*と等しい文字列を探索
    - もし、*である場合
        - その一つ前の文字を削除
    - もし、それ以外である場合
        - 新しい文字列resultに追加
・resultを返す
 */

File: LeetCode75/2390_Removing_Stars_From_a_String/2390_Removing_Stars_From_a_String.kt
class Solution {
    fun removeStars(s: String): String {
        var result = StringBuilder()

        for (i in 0 until s.length) {
            if (s[i] === '*') {
                if (result.isNotEmpty()) {
                    result.deleteCharAt(result.length - 1)
                }
            } else {
                result.append(s[i])
            }
        }
        return result.toString()
    }
}
/*
【アプローチ】
・*の前にある文字を削除する
・*を削除する
・最終的に出力される文字を返す
・*を見つける
・*の前の文字を指定
・*の前の文字は、新しく作成された箱に入らない

【ミニマム】
*を見つけ出して、それより1つ前の文字を削除すること
・全探索を行う
- ループ処理
    - for (i in s.length)
・*を見つける
- if文
    - if (i == '*')
    - iの一つ前の文字をresultの最後の文字から取得
        - resultの最後文字を取得
            - この取得の方法がわからない
        - それを削除
        - resultから最後の文字が消えて、新しいresultとなる
・*以外の文字列
- resultに結果を返す値を格納
・結果を返す
 */
